numpy/lib/arraypad.py

```diff
@@ -69,821 +81,475 @@ def _round_ifneeded(arr, dtype):
 
-    # Concatenate `arr` with `med_chunk`, extended along `axis` by `pad_amt`
-    return _do_append(arr, med_chunk.repeat(pad_amt, axis), axis=axis)
+    return left_ramp, right_ramp
 
 
-def _prepend_min(arr, pad_amt, num, axis=-1):
+def _get_stats(padded, axis, index_pair, length_pair, stat_func):
     """
-    Prepend `pad_amt` minimum values along `axis`.
+    Calculate statistic for the empty-padded array in given dimnsion.
 
     Parameters
     ----------
-    arr : ndarray
-        Input array of arbitrary shape.
-    pad_amt : int
-        Amount of padding to prepend.
-    num : int
-        Depth into `arr` along `axis` to calculate minimum.
-        Range: [1, `arr.shape[axis]`] or None (entire axis)
+    padded : ndarray
+        Empty-padded array.
     axis : int
-        Axis along which to pad `arr`.
+        Dimension in which the statistic is calculated.
+    index_pair : (int, int)
+        Pair of indices that mark the end (or start) of the pad area on both
+        sides in the given dimension.
+    length_pair : 2-element sequence of None or int
+        Gives the number of values in valid area from each side that is
+        taken into account when calculating the statistic. If None the entire
+        valid area in `padded` is considered.
+    stat_func : function
+        Function to compute statistic. The expected signature is
+        ``stat_func(x: ndarray, axis: int, keepdims: bool) -> ndarray``.
 
     Returns
     -------
-    padarr : ndarray
-        Output array, with `pad_amt` values prepended along `axis`. The
-        prepended region is the minimum of the first `num` values along
-        `axis`.
-
-    """
-    if pad_amt == 0:
-        return arr
-
-    # Equivalent to edge padding for single value, so do that instead
-    if num == 1:
-        return _prepend_edge(arr, pad_amt, axis)
-
-    # Use entire array if `num` is too large
-    if num is not None:
-        if num >= arr.shape[axis]:
-            num = None
-
-    # Slice a chunk from the edge to calculate stats on
-    min_slice = _slice_first(arr.shape, num, axis=axis)
-
-    # Extract slice, calculate min
-    min_chunk = arr[min_slice].min(axis=axis, keepdims=True)
+    left_stat, right_stat : ndarray
+        Calculated statistic for both sides of `padded`.
+    """
+    max_length = padded.shape[axis] - index_pair[0] - index_pair[1]
+
+    if index_pair[0] > 0:
+        if length_pair[0] is not None and length_pair[0] < max_length:
+            left_length = length_pair[0]
+        else:
+            left_length = max_length
+        left_slice = _slice_at_axis(
+            padded.shape,
+            slice(index_pair[0], index_pair[0] + left_length),
+            axis
+        )
+        left_chunk = padded[left_slice]
+        left_stat = stat_func(left_chunk, axis=axis, keepdims=True)
+        _round_if_needed(left_stat, padded.dtype)
+    else:
+        left_stat = np.array([], dtype=padded.dtype)
```

Comment:
I agree. This function is easier to reason about without this edge case.

We could avoid calling this function in the first place. Meaning we'd check for `pad_width == 0` before calling this function. But that will only work if both sides are padded with 0.

Reply:
Maybe it really is best to create just slice an array of 0 size out of the existing one and return it. I mean all that is needed is `padded[_slice_at_axis(axis, slice(0, 0))]` or so?

Reply:
Yeah, that would work. I'll think about it. If we can save computation time like suggested in https://github.com/numpy/numpy/pull/11358#discussion_r244536680 it might be worth it to skip the optimization for the "0 path".